main/lsp/requests/rename.cc

```diff
@@ -44,12 +44,24 @@ bool isSubclass(const core::GlobalState &gs, core::SymbolRef root, core::SymbolR
     }
     visited[s.classOrModuleIndex()] = true;
 
-    auto super = s.data(gs)->superClass();
-    if (super.exists()) {
-        memoized[s.classOrModuleIndex()] = isSubclass(gs, root, super, memoized, visited);
+    if (s.data(gs)->derivesFrom(gs, root)) {
+        memoized[s.classOrModuleIndex()] = true;
     } else {
-        memoized[s.classOrModuleIndex()] = false;
+        auto super = s.data(gs)->superClass();
+        auto parents = s.data(gs)->mixins();
+        // Reverse parents so the last loaded mixin is first
```

Comment:
Mixins should already be in resolution order, although I'm not sure if order is important here.